lib/mshoplib/src/MShop/Plugin/Provider/Order/PropertyAdd.php

```diff
@@ -199,6 +179,12 @@ protected function getProductItems( iterable $productIds, iterable $productCodes
 			$search->is( 'product.code', '==', $productCodes ),
 		] ) );
 
-		return $manager->search( $search, ['product/property'] );
+		$items = $manager->search( $search, ['product/property'] );
+
+		return map( $types )->map( function ( $type ) use ( $items ) {
+			if( !( $properties = $items->getProperties($type)->collapse() )->empty() ) {
+				return [ $type => $properties ];
+			}
+		} )->filter()->collapse(1);
```

Comment:
This should be sufficient to get a product code index and the properties as list of items:
```
return $manager->search( $search, ['product/property'] )->col( null, 'product.code' )->getProperties( $types );
```

Reply:
`getProperties` requires a string not an array, does it accept comma separated values? If so we would need to implode `$types`.